xmtp_db/src/encrypted_store/group_message.rs

```diff
@@ -317,7 +317,48 @@ impl<C: ConnectionExt> DbConnection<C> {
             query = query.limit(limit);
         }
 
-        self.raw_query_read(|conn| query.load::<StoredGroupMessage>(conn))
+        let messages = self.raw_query_read(|conn| query.load::<StoredGroupMessage>(conn))?;
+
+        // Check if this is a DM group
+        let is_dm = self.raw_query_read(|conn| {
+            groups_dsl::groups
+                .filter(groups_dsl::id.eq(group_id))
+                .select(groups_dsl::conversation_type)
+                .first::<ConversationType>(conn)
+        })? == ConversationType::Dm;
+
+        // If DM, retain only one GroupUpdated message (the oldest)
+        let messages = if is_dm && args.limit.is_none() {
```

Comment:
Nit: since this is on the query messages code, it would be calculated every time, for now I would say it's ok, but we need to think about this kind of complexities to not make message querying slower.

Reply:
💯 I thought about this as well. Probably something we need to think about eventually